proposals/ghci-clear-command.md

```diff
@@ -0,0 +1,86 @@
+---
+author: Cate Roxl
+date-accepted: ""
+ticket-url: ""
+implemented: ""
+---
+
+This proposal is [discussed at this pull request](https://github.com/ghc-proposals/ghc-proposals/pull/382).
+
+
+# Make learning and playing with Haskell through GHCi easier through adding `:clear [variable identifier]`.  
+
+Adding :clear [variable identifier] that only applies to variable identifiers or functions defined in GHCi.
+
+## Motivation
+
+Haskell, as purely functional language, has been catching the attention of newcomers and experienced programmers who are wanting to learn Functional Programming/Thinking.
+
+While I acknowledge immutability being the central concept of functional programming—perhaps it may be the reason why `:clear` doesn't exist, the proposed GHCi command may just make interacting with GHCi more convenient; and learning Haskell more fun.
+
+
+## Proposed Change Specification
+
+Let's say that we're taking a Haskell course, and we're asked to write a function to reverse a list. So we start up GHCi, then we go:
+
+```
+Prelude> rev x = reverse x
+Prelude> rev "Haskell"
+"lleksaH"
+```
+
+Now there are other ways to write such function, which we can play and study with. But because of immutability, once we declare `rev = reverse x` or any variable identifier to a variable: it stays that way. 
+
+Unless. . . We quit GHCi and start it up again. 
```

Comment:
Does it really stay that way? GHCi allows you to shadow earlier bindings:
```
Prelude> foo x = x
Prelude> foo ()
()
Prelude> foo x = x + 1
Prelude> foo 1
2
```

Of course it can still be confusing that the old function is still around:
```
Prelude> foo x = x + 1
Prelude> foo 1
2
Prelude> bar x = foo x
Prelude> bar 1
2
Prelude> foo x = x + 2
Prelude> foo 1
3
Prelude> bar 1
2
```
but I that is, even maybe initially confusing, explainable.



```

Reply:
Oh! I tried it and it works. Thank you for this. There are statements in my proposal that are wrong. It doesn't stay that way as I thought. It may just be the books and tutorials that I've come across when they discuss immutability. It's my mistake for not considering that the GHCi allows changes in definition. I've always thought of it being immutable right there in the GHCi.

Reply:
I'm closing the pull request. Thank you once again.